pkg/subctl/cmd/utils/restconfig/restconfig.go

```diff
@@ -0,0 +1,162 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package restconfig
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+
+	"github.com/submariner-io/admiral/pkg/resource"
+	subv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+
+	"github.com/submariner-io/submariner-operator/apis/submariner/v1alpha1"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/utils"
+)
+
+type RestConfig struct {
+	Config      *rest.Config
+	ClusterName string
+}
+
+func GetMultipleRestConfigs(kubeConfigPath string, kubeContexts []string) ([]RestConfig, error) {
+	var restConfigs []RestConfig
+
+	rules := clientcmd.NewDefaultClientConfigLoadingRules()
+	rules.DefaultClientConfig = &clientcmd.DefaultClientConfig
+	overrides := &clientcmd.ConfigOverrides{ClusterDefaults: clientcmd.ClusterDefaults}
+	rules.ExplicitPath = kubeConfigPath
+
+	contexts := []string{}
+	if len(kubeContexts) > 0 {
+		contexts = append(contexts, kubeContexts...)
+	} else {
+		kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(rules, overrides)
+		rawConfig, err := kubeConfig.RawConfig()
+		if err != nil {
+			return restConfigs, err
+		}
+		for context := range rawConfig.Contexts {
+			contexts = append(contexts, context)
+		}
+	}
+
+	for _, context := range contexts {
+		if context != "" {
+			overrides.CurrentContext = context
+			config, err := GetClientConfigAndClusterName(rules, overrides)
+			if err != nil {
+				return nil, err
+			}
+
+			restConfigs = append(restConfigs, config)
+		}
+	}
+
+	return restConfigs, nil
+}
+
+func GetBrokerRestConfigAndNamespace(submariner *v1alpha1.Submariner,
```

Comment:
```suggestion
func ForBroker(submariner *v1alpha1.Submariner,
```

Would look better restconfig.ForBroker.
Also the module is already called restconfig so repeating it here is stuttering :)